src/UI/Text.elm

```diff
@@ -99,50 +96,42 @@ overline content =
     defaultText SizeOverline content
 
 
-colorPrimary : TextColor
-colorPrimary =
-    Internal.ColorPalette
-        ( Palette.tonePrimary, Palette.lumMiddle )
-
-
-colorBgMiddle : TextColor
-colorBgMiddle =
-    Internal.ColorBgMiddle
-
-
-colorInverted : TextColor
-colorInverted =
-    Internal.ColorInverted
-
-
-colorGray : TextColor
-colorGray =
-    Internal.ColorPalette
-        ( Palette.toneGray, Palette.lumMiddle )
-
-
-colorFromPalette : Palette.Color -> TextColor
-colorFromPalette color =
-    Internal.ColorPalette color
-
-
-withColor : TextColor -> Text -> Text
-withColor color (Internal.Text prop opt) =
-    Internal.Text prop { opt | color = color }
+withColor : Palette.Color -> Text -> Text
+withColor color text =
+    mapOpt (\opt -> { opt | color = Internal.ColorPalette color }) text
 
 
 withEllipsis : Bool -> Text -> Text
-withEllipsis val (Internal.Text prop opt) =
-    Internal.Text prop { opt | oneLineEllipsis = val }
+withEllipsis val text =
+    mapOpt (\opt -> { opt | oneLineEllipsis = val }) text
 
 
 toEl : RenderConfig -> Text -> Element msg
-toEl cfg (Internal.Text { content, size } { color, oneLineEllipsis }) =
-    let
-        attrs =
-            Internal.attributes cfg size oneLineEllipsis color
-    in
-    content
-        |> Element.text
-        |> List.singleton
-        |> Element.paragraph attrs
+toEl cfg (Internal.Text spans) =
+    case spans of
+        [] ->
+            Element.none
+
+        [ theOne ] ->
+            Internal.spanRenderEl cfg theOne
+
+        _ ->
+            -- TODO: Concat paragraphs
+            List.map (Internal.spanRenderEl cfg) spans
+                |> Element.column [ Element.spacing 0 ]
+
+
+multiline : (String -> Text) -> List String -> Text
+multiline style lines =
+    lines
+        |> List.map (style >> Internal.getSpans)
+        |> List.concat
+        |> Internal.Text
+
+
+combination : List Text -> Text
```

Comment:
what is the use case for that?

Reply:
elm-ui allows styling first in text elements, then paragraph elements. But, as we use line-height, our smallest unit is a paragraph. By merging Texts together we can wisely style the text element. This would mainly allow:
```plain
Start some text like this, THEN WOW, THIS IS OVERLINED! But turns back normal! (Small captions at the end).
```

But would also allow us to space text elements differently in the same text, like in Action Bar we have
```
SOME TITLE WITH BODY1 SIZE
Some Caption
```

We can make that every time this happens he have the correct spacing btw them.